CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesByLocationNGTest.java

```diff
@@ -0,0 +1,338 @@
+     * missing LatitudeAttribute
+     */
+    @Test
+    public void testGetNodesToMerge_NoLatitudeAttribute() throws Exception {
+        System.out.println("testGetNodesToMerge_NoLatitudeAttribute");
+        Comparator<String> leadVertexChooser = null;
+        int threshold = 1600000;
+        boolean selectedOnly = false;
+        MergeNodesByLocation instance = new MergeNodesByLocation();
+        Map<Integer, Set<Integer>> expResult = new HashMap<>();
+
+        // create a new analytic graph
+        final Schema schema = SchemaFactoryUtilities.getSchemaFactory(AnalyticSchemaFactory.ANALYTIC_SCHEMA_ID).createSchema();
+        StoreGraph graph = new StoreGraph(schema);
+
+        // add attributes
+        int vertexIdentifierAttribute = VisualConcept.VertexAttribute.IDENTIFIER.ensure(graph);
+
+        // add vertices
+        int vxId1 = graph.addVertex();
+        int vxId2 = graph.addVertex();
+        int vxId3 = graph.addVertex();
+
+        // set the identifier of each vertex to something unique
+        graph.setStringValue(vertexIdentifierAttribute, vxId1, "V1");
+        graph.setStringValue(vertexIdentifierAttribute, vxId2, "V2");
+        graph.setStringValue(vertexIdentifierAttribute, vxId3, "V3");
+
+        Map result = instance.getNodesToMerge(graph, leadVertexChooser, threshold, selectedOnly);
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of getNodesToMerge method, of class MergeNodesByLocation, with
+     * selectedOnly = False
+     */
+    @Test
+    public void testGetNodesToMergee_Threshold_1600000_selectedOnlyFalse_1() throws Exception {
+        System.out.println("testGetNodesToMergee_Threshold_1600000_selectedOnlyFalse_1");
+        Comparator<String> leadVertexChooser = null;
+        int threshold = 1600000;
+        boolean selectedOnly = false;
+        MergeNodesByLocation instance = new MergeNodesByLocation();
+
+        // set the Latitude of each vertex to something unique
+        graph.setStringValue(vertexLatitudeAttribute, vxId1, "15");
+        graph.setStringValue(vertexLatitudeAttribute, vxId2, "25");
+        graph.setStringValue(vertexLatitudeAttribute, vxId3, "37");
+        graph.setStringValue(vertexLatitudeAttribute, vxId4, "26");
+        graph.setStringValue(vertexLatitudeAttribute, vxId5, "38");
+
+        // set the Longitude of each vertex to something unique
+        graph.setStringValue(vertexLongitudeAttribute, vxId1, "35");
+        graph.setStringValue(vertexLongitudeAttribute, vxId2, "45");
+        graph.setStringValue(vertexLongitudeAttribute, vxId3, "56");
+        graph.setStringValue(vertexLongitudeAttribute, vxId4, "46");
+        graph.setStringValue(vertexLongitudeAttribute, vxId5, "57");
+
+        Map<Integer, Set<Integer>> expResult = new HashMap<>();
+        Set<Integer> cluster1 = new HashSet<>();
+        cluster1.addAll(Arrays.asList(new Integer[]{0, 1, 2, 3, 4}));
```

Comment:
You can specify the elements of `Arrays.asList()` without needing to declare an array as an FYI (i.e. `Arrays.asList(0, 1, 2, 3, 4)` also works)